core/utils.py

```
@@ -463,3 +464,48 @@ def download_image_from_url(
+    Returns:
+        A list of floats representing the embedding vector, or None if the request fails
+    """
+    url = "https://api.jina.ai/v1/embeddings"
+    headers = {
+        "Content-Type": "application/json",
+        "Authorization": f"Bearer {settings.JINA_READER_API_KEY}",
+    }
+    data = {"model": "jina-embeddings-v3", "task": "text-matching", "input": [text]}
+
+    try:
+        response = requests.post(url, headers=headers, json=data)
```

Ruff (0.14.3)

Probable use of `requests` call without timeout

(S113)